clusterrunner_ha

```diff
@@ -0,0 +1,309 @@
+#!/bin/bash
+# clusterrunner_ha - start, stop, and restart ClusterRunner in HA mode.#
+#
+# description: clusterrunner_ha is a wrapper around the ClusterRunnner
+# services with an nginx proxy in front for automatic failover.
+
+CONF_PATH=/etc/sysconfig/clusterrunner_ha
+test -e "$CONF_PATH" || { echo "$CONF_PATH doesn't exist"; exit 1; }
+test -r "$CONF_PATH" || { echo "$CONF_PATH not readable. Perhaps you forgot 'sudo'?"; exit 1; }
+source "$CONF_PATH"
+
+##################################################################################
+# Execute commmand as the jenkins LDAP user on target remote.
+# This is usually necessary in order to utilize the jenkins user's ssh keys.
+#
+# Args:
+#   TARGET str - Remote host name.
+#   COMMAND str - Command to execute on remote.
+#   IS_ASYNC bool (optional) - Should this remote command be async? Default: false.
+##################################################################################
+run_as_jenkins_on_remote()
+{
+    # Gather bash function arugments
+    TARGET=$1
+    COMMAND=$2
+    IS_ASYNC=${3:-false}
+
+    # Build ssh/scp command
+    SSH_COMMAND="ssh -o ConnectTimeout=5 -o BatchMode=yes -o PasswordAuthentication=no -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no $TARGET \"$COMMAND\""
+    if $IS_ASYNC
+    then
+        SSH_COMMAND="$SSH_COMMAND &"
+    fi
+
+    # Execute remote command as jenkins
+    sudo su jenkins -c "$SSH_COMMAND"
```

Comment:
Obviously, replace 'jenkins' with whatever user you are using to run ClusterRunner.